daml-lf/archive/src/main/protobuf/com/daml/daml_lf_dev/daml_lf2.proto

```diff
@@ -1731,12 +1691,8 @@ message DefDataType {
   }
 
   // name of the defined data type
-  oneof name {
-    DottedName name_dname = 1;
-
-    // *Must be a valid interned name*
+  // *Must be a valid interned name*
```

Comment:
without the `DottedName` alternative this becomes a bit confusing, I'd appreciate if you could update these comments to match whether the int indexes into `interned_strings` or `interned_dotted_names`

```suggestion
  // *Must be a valid interned dotted name*
```

Reply:
On a related note, perhaps it would make sense to define this as its own message for type safety? The indirection would make our dalf's larger but maybe not too much?

```proto
message InternedDottedName {
  int32 index = 1;
}

...

message EnumConstructors { 
  ...
  InternedDottedName name_interned_dname = 8;
}
```

Reply:
The original confusion comes from the spec that use Name for DottedName and Identifier for Name.
Maybe a good idea to fix this mismatch, but for now I will continue to use the spec convention.

Reply:
Concerning the type safety. The interning was done (by Martin Huschenbett) to save space.  Taking into we have something working fine for years, I am a bit reluctant to de-optimize the choices he made.